src/core/executors/SubscriptionExecutor.ts

```diff
@@ -12,6 +12,10 @@ export class SubscriptionExecutor extends ExecutorBase {
   }
 
   processDeltaQueue(): void {
+    if (this._deltaQueue.length === 0) {
```

Comment:
Length check seems unnecessary, can we remove it?

Reply:
I think we can remove it. I was copying the pattern in the other executors: [IdentityExecutor](https://github.com/OneSignal/OneSignal-Website-SDK/blob/main/src/core/executors/IdentityExecutor.ts#L16), [PropertiesExecutor](https://github.com/OneSignal/OneSignal-Website-SDK/blob/ff52c3cd26b2c90caf9c7af80883f4f5503acafe/src/core/executors/PropertiesExecutor.ts#L14). Do you think those could be removed as well?

Reply:
Probably good to avoid doing unnecessary work

Reply:
I think we can remove it since the the next line will run `this._deltaQueue.forEach(...)` and since the delta queue is empty, it wouldn't do any work anyway. Technically I think the only extra work it would do is to create a variable but I think that is negligible.